### PR TITLE
Various minor cleanups: Remove redundant or dead code. Avoid UB. Remove unused parameters.

### PR DESCRIPTION
**Pull request by practicalswift**
Various minor cleanups: Remove dead or dead code. Avoid UB. Remove unused parameters.

### CODE REVIEW

File: bitcoin/test/run-tx-encode.c [hunk 1]
```diff
@@ -12,7 +12,7 @@ static void hexeq(const void *p, size_t len, const char *hex)
 {
 	char *tmphex = tal_hexstr(NULL, p, len);
 
-	if (strcmp(hex, tmphex)) {
+	if (strcmp(hex, tmphex) != 0) {
```
Comment (ZmnSCPxj) [inline bitcoin/test/run-tx-encode.c]: `ccan/str/str.h` has a `streq`, might be better here (unless this test does not want to pull in ccan?)

File: lightningd/pay.c [hunk 2]
```diff
@@ -370,7 +370,6 @@ void payment_failed(struct lightningd *ld, const struct htlc_out *hout,
 		    const char *localfail)
 {
 	struct onionreply *reply;
-	enum onion_type failcode;
```
Comment (ZmnSCPxj) [inline lightningd/pay.c]: good catch!

File: bitcoin/pullpush.c [hunk 3]
```diff
@@ -80,7 +80,7 @@ u64 pull_varint(const u8 **cursor, size_t *max)
 
 u32 pull_le32(const u8 **cursor, size_t *max)
 {
-	le32 ret;
+	le32 ret = 0;
```
Comment (ZmnSCPxj) [inline bitcoin/pullpush.c]: I think unnecessary? If `pull` succeeds the entire variable gets overwritten.  Similar `pull_le64`.

File: bitcoin/signature.c [hunk 4]
```diff
@@ -197,7 +197,7 @@ static bool IsValidSignatureEncoding(const unsigned char sig[], size_t len)
 
     // Verify that the length of the signature matches the sum of the length
     // of the elements.
-    if ((size_t)(lenR + lenS + 7) != len) return false;
+    if ((size_t)lenR + (size_t)lenS + 7 != len) return false;
```
Comment (ZmnSCPxj) [inline bitcoin/signature.c]: Yes, definitely safer this way, good catch.

**Comment by ZmnSCPxj** [comment]
ACK f0bb3fb318e26c6e04b12fd05ab0cad552acfeb1

File: bitcoin/signature.c (hunk 4 above)
Comment (rustyrussell) [reply bitcoin/signature.c]: Interestingly, this is cut & paste from the bitcoind source :)

File: lightningd/onchain_control.c [hunk 5]
```diff
@@ -46,7 +46,7 @@ static void onchaind_tell_fulfill(struct channel *channel)
 	}
 }
 
-static void handle_onchain_init_reply(struct channel *channel, const u8 *msg)
+static void handle_onchain_init_reply(struct channel *channel)
 {
```
Comment (rustyrussell) [inline lightningd/onchain_control.c]: Correct fix here is to actually fromwire_onchain_init_reply() the message, I think.  That's more uniform and more future proof.

File: lightningd/onchain_control.c [hunk 6]
```diff
@@ -202,7 +202,7 @@ static void handle_onchain_htlc_timeout(struct channel *channel, const u8 *msg)
 	onchain_failed_our_htlc(channel, &htlc, "timed out");
 }
 
-static void handle_irrevocably_resolved(struct channel *channel, const u8 *msg)
+static void handle_irrevocably_resolved(struct channel *channel)
```
Comment (rustyrussell) [inline lightningd/onchain_control.c]: Here too...

File: bitcoin/pullpush.c [hunk 7]
```diff
@@ -58,7 +58,7 @@ const u8 *pull(const u8 **cursor, size_t *max, void *copy, size_t n)
 	}
 	*cursor += n;
 	*max -= n;
-	if (copy)
+	if (copy && p)
```
Comment (rustyrussell) [inline bitcoin/pullpush.c]: Would rather test `n` here so we'll get a nice crash if `p` is NULL somehow (shouldn't happen: *max will be 0 in that case, so `n` must be zero).

File: bitcoin/pullpush.c (hunk 3 above)
Comment (rustyrussell) [reply bitcoin/pullpush.c]: Even if it *fails* it memsets the whole thing to zero, for safety.

File: bitcoin/signature.c (hunk 4 above)
Comment (practicalswift) [reply bitcoin/signature.c]: I reported it upstream here: https://github.com/bitcoin/bitcoin/pull/12611 :-)

**Comment by practicalswift** [comment]
Updated in light of reviews. Please re-review :-)

**Comment by ZmnSCPxj** [comment]
ACK 7a09ebdd18dc3db11b7f7f7b22baefdaab8a1c2d

File: bitcoin/pullpush.c [hunk 8]
```diff
@@ -1,3 +1,5 @@
+#include <assert.h>
```
Comment (ZmnSCPxj) [inline bitcoin/pullpush.c]: CI complains about this, it should be before the `#include <ccan/endian/endian.h>`

**Comment by practicalswift** [comment]
@ZmnSCPxj Fixed! Please re-review :-)

**Comment by ZmnSCPxj** [comment]
ACK 0d51b2d20d34882d4b333d03d5ee9077e2cd7390

**Comment by practicalswift** [comment]
Rebased! :-)

**Pull request merged**